src/main/java/authentication/jwt/AuthenticationFilter.java

```diff
@@ -52,15 +52,13 @@ public void doFilter(ServletRequest servletRequest, ServletResponse servletRespo
         HttpServletResponse response = (HttpServletResponse) servletResponse;
 
         if (matchesFilteredPattern(request.getServletPath())) {
-            String jwt =
-                request.getHeader(AUTHORIZATION_HEADER)
-                    .substring(AUTHORIZATION_HEADER_BEARER_PREFIX.length());
-            if (jwt == null) {
+            String jwt = request.getHeader(AUTHORIZATION_HEADER);
+            if (jwt == null || jwt.length() < AUTHORIZATION_HEADER_BEARER_PREFIX.length()) {
```

Comment:
Could you explain statement `jwt.length() < AUTHORIZATION_HEADER_BEARER_PREFIX.length()`?

Reply:
I added the condition to avoid IndexOutOfBoundException can be thrown by the substring below.